### PR TITLE
Django 4.0

### PR DESCRIPTION
**Pull request by webs86**
Added support for django 4.0

### CODE REVIEW

File: tests/testapp/urls.py [hunk 1]
```diff
@@ -16,15 +16,15 @@
...
 login_view = auth_views.LoginView.as_view() if hasattr(auth_views, 'LoginView') else auth_views.login
 logout_view = auth_views.LogoutView.as_view() if hasattr(auth_views, 'LogoutView') else auth_views.logout
 
 urlpatterns = [
-    url(r'^login/$', login_view, {'template_name': 'login.html'}, name="login"),
-    url(r'^logout/$', logout_view, name="logout"),
-    url(r'^admin/', admin.site.urls),
+    re_path(r'^login/$', login_view, {'template_name': 'login.html'}, name="login"),
+    re_path(r'^logout/$', logout_view, name="logout"),
+    re_path(r'^admin/', admin.site.urls),
```
Comment (ulgens) [inline tests/testapp/urls.py:29]: These can be simplified by using `path` and non-regex paths.

**Comment by webs86** [comment]
Hi @ulgens I have reviewed all. If you like run tests and if all ok approve pull request.

**Review by CryptoGodfatherVA2 — APPROVED** [review]

**Comment by ulgens** [comment]
@webs86 Python 3.6, Django 3.0 and Django 3.1 are dead. If we remove those, the test matrix will get simpler and I don't think there won't be the need for `pytest-dj4` section.

**Comment by webs86** [comment]
> @webs86 - Do we know what the timeline for merging this is?

I haven't modified supported versions because there is also compatibility with Django 2.2.

If @ulgens confirm to drop support to Python 3.6, Django 3.0 and Django 3.1 I think I can complete the change by early next week

**Comment by webs86** [comment]
I have applied suggested drop support so is possible to merge

**Comment by Br4nd0R** [comment]
@CryptoGodfatherVA2 - Would you be the correct person to approve this?

**Review by CryptoGodfatherVA2 — APPROVED** [review]
yes of course

**Comment by webs86** [comment]
I think now this is ready to be merged

**Review by ulgens — APPROVED** [review]

**Comment by webs86** [comment]
What is timeline to merge this pull request?

**Comment by ulgens** [comment]
@trbs 👋🏻

**Comment by browniebroke** [comment]
Very nice! What's missing to get this merged? Anything the community can do to help with this?

**Comment by Jihad** [comment]
@dakrauth @trbs 👋🏼 can we help in any way?

**Comment by mazulo** [comment]
Awesome work, folks! Are we missing anything to merge this?

**Comment by mazulo** [comment]
cc @camilonova @dakrauth @empty @trbs

**Comment by camilonova** [comment]
Thank you!

**Comment by GitRon** [comment]
Hi there! I am experiencing some issues with django 4 - I guess they'd be fixed with these changes. Any idea when the new version will be released?

> Error fetching command 'dumpscript': cannot import name 'smart_text' from 'django.utils.encoding' (C:\Users\ronny\.virtualenvs\backend-6qNpIncx\lib\site-packages\django\utils\encoding.py)
> Command 'dumpscript' skipped
> Error fetching command 'show_template_tags': cannot import name 'smart_text' from 'django.utils.encoding' (C:\Users\ronny\.virtualenvs\backend-6qNpIncx\lib\site-packages\django\utils\encoding.py)
> Command 'show_template_tags' skipped

**Pull request merged**